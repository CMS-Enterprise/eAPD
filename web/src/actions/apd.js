import { push } from 'connected-react-router';

import { getIsAdmin } from '../reducers/user.selector';
import axios from '../util/api';
import { fromAPI, toAPI } from '../util/serialization/apd';

const LAST_APD_ID_STORAGE_KEY = 'last-apd-id';

export const ADD_APD_KEY_PERSON = 'ADD_APD_KEY_PERSON';
export const CREATE_APD = 'CREATE_APD';
export const CREATE_APD_REQUEST = 'CREATE_APD_REQUEST';
export const CREATE_APD_SUCCESS = 'CREATE_APD_SUCCESS';
export const CREATE_APD_FAILURE = 'CREATE_APD_FAILURE';
export const DELETE_APD_REQUEST = Symbol('delete apd : request');
export const DELETE_APD_SUCCESS = Symbol('delete apd : success');
export const DELETE_APD_FAILURE = Symbol('delete apd : failure');
export const GET_APD_REQUEST = 'GET_APD_REQUEST';
export const GET_APD_SUCCESS = 'GET_APD_SUCCESS';
export const GET_APD_FAILURE = 'GET_APD_FAILURE';
export const REMOVE_APD_KEY_PERSON = 'REMOVE_APD_KEY_PERSON';
export const SAVE_APD_REQUEST = 'SAVE_APD_REQUEST';
export const SAVE_APD_SUCCESS = 'SAVE_APD_SUCCESS';
export const SAVE_APD_FAILURE = 'SAVE_APD_FAILURE';
export const SELECT_APD = 'SELECT_APD';
export const SUBMIT_APD_REQUEST = 'SUBMIT_APD_REQUEST';
export const SUBMIT_APD_SUCCESS = 'SUBMIT_APD_SUCCESS';
export const SUBMIT_APD_FAILURE = 'SUBMIT_APD_FAILURE';
export const UPDATE_APD = 'UPDATE_APD';
export const UPDATE_BUDGET = 'UPDATE_BUDGET';
export const WITHDRAW_APD_REQUEST = Symbol('withdraw apd request');
export const WITHDRAW_APD_SUCCESS = Symbol('withdraw apd success');
export const WITHDRAW_APD_FAILURE = Symbol('withdraw apd failure');

export const SET_SELECT_APD_ON_LOAD = 'SET_SELECT_APD_ON_LOAD';
export const selectApdOnLoad = () => (dispatch, getState) => {
  const isAdmin = getIsAdmin(getState());
  if (!isAdmin) {
    dispatch({ type: SET_SELECT_APD_ON_LOAD });
  }
};

export const addKeyPerson = () => ({ type: ADD_APD_KEY_PERSON });
export const removeKeyPerson = (key, { global = window } = {}) => dispatch => {
  if (global.confirm('Do you really want to delete this key person?')) {
    dispatch({
      type: REMOVE_APD_KEY_PERSON,
      key
    });
  }
};

export const updateBudget = () => (dispatch, getState) =>
  dispatch({ type: UPDATE_BUDGET, state: getState() });

export const requestApd = () => ({ type: GET_APD_REQUEST });
export const receiveApd = data => ({ type: GET_APD_SUCCESS, data });
export const failApd = error => ({ type: GET_APD_FAILURE, error });

export const updateApd = updates => dispatch => {
  dispatch({ type: UPDATE_APD, updates });
  if (updates.years) {
    dispatch(updateBudget());
  }
};

export const selectApd = (
  id,
  { deserialize = fromAPI, global = window, pushRoute = push } = {}
) => dispatch =>
  axios.get(`/apds/${id}`).then(req => {
    dispatch({ type: SELECT_APD, apd: deserialize(req.data) });
    dispatch(updateBudget());
    dispatch(pushRoute('/apd'));

    if (global.localStorage) {
      global.localStorage.setItem(LAST_APD_ID_STORAGE_KEY, id);
    }
  });

export const createRequest = () => ({ type: CREATE_APD_REQUEST });
export const createSuccess = data => ({ type: CREATE_APD_SUCCESS, data });
export const createFailure = () => ({ type: CREATE_APD_FAILURE });
export const createApd = ({
  deserialize = fromAPI,
  pushRoute = push
} = {}) => dispatch => {
  dispatch(createRequest());
  return axios
    .post('/apds')
    .then(async req => {
      const apd = deserialize(req.data);
      dispatch(createSuccess(apd));
      await dispatch(selectApd(apd.id, { deserialize, pushRoute }));
    })
    .catch(error => {
      const reason = error.response ? error.response.data : 'N/A';
      dispatch(createFailure(reason));
    });
};

export const requestSave = () => ({ type: SAVE_APD_REQUEST });
export const saveSuccess = data => ({ type: SAVE_APD_SUCCESS, data });
export const saveFailure = data => ({ type: SAVE_APD_FAILURE, data });

export const fetchApd = ({
  global = window,
  pushRoute = push,
  select = selectApd
} = {}) => (dispatch, getState) => {
  dispatch(requestApd());

  const url = `/apds`;

  return axios
    .get(url)
    .then(async req => {
      const apd = Array.isArray(req.data) ? req.data : null;
      dispatch(receiveApd(apd.filter(({ status }) => status !== 'archived')));

      const {
        apd: { selectAPDOnLoad }
      } = getState();
      if (
        selectAPDOnLoad &&
        global.localStorage &&
        global.localStorage.getItem(LAST_APD_ID_STORAGE_KEY)
      ) {
        await dispatch(
          select(global.localStorage.getItem('last-apd-id'), {
            global,
            pushRoute
          })
        );
      }
    })
    .catch(error => {
      const reason = error.response ? error.response.data : 'N/A';
      dispatch(failApd(reason));
    });
};

const shouldFetchApd = state => !state.apd.loaded;

export const fetchApdDataIfNeeded = () => (dispatch, getState) => {
  if (shouldFetchApd(getState())) {
    return dispatch(fetchApd());
  }

  return null;
};

export const saveApd = ({ serialize = toAPI } = {}) => (dispatch, state) => {
  const {
    apd: { data: updatedApd },
    activities,
    dirty
  } = state();

  if (!dirty.dirty) {
    return Promise.resolve();
  }

  dispatch(requestSave());

  const apd = serialize(updatedApd, activities);

  // These are the fields we want to remove if they aren't dirty.
  const filterAPDFields = [
    'federalCitations',
    'incentivePayments',
    'narrativeHIE',
    'narrativeHIT',
    'narrativeMMIS',
    'programOverview',
    'previousActivityExpenses',
    'previousActivitySummary',
    'stateProfile'
  ];
  const filterActivityFields = [
    'contractorResources',
    'costAllocation',
    'costAllocationNarrative',
    'expenses',
    'goals',
    'schedule',
    'standardsAndConditions',
    'statePersonnel',
    'quarterlyFFP'
  ];

  filterAPDFields.forEach(field => {
    if (!dirty.data.apd[field]) {
      delete apd[field];
    }
  });

  for (let i = apd.activities.length - 1; i >= 0; i -= 1) {
    const activity = apd.activities[i];
    if (!dirty.data.activities.byKey[activity.key]) {
      // If this activity isn't dirty, strip it from the list to save,
      // but then add just its ID back - this prevents it from being
      // deleted by the API synchronization.
      apd.activities.splice(i, 1);
      apd.activities.push({ id: activity.id });
    } else {
      const dirtyActivity = dirty.data.activities.byKey[activity.key];
      // Get rid of any activity fields that aren't dirty.
      filterActivityFields.forEach(field => {
        if (!dirtyActivity[field]) {
          delete activity[field];
        }
      });
    }
  }

  return axios
    .put(`/apds/${updatedApd.id}`, apd)
    .then(res => {
      dispatch(saveSuccess(res.data));
      return res.data;
    })
    .catch(error => {
      if (error.response.status === 403) {
        dispatch(saveFailure('save-apd.not-logged-in'));
      } else {
        dispatch(saveFailure());
      }
      throw error;
    });
};

export const deleteApd = (id, { fetch = fetchApd } = {}) => dispatch => {
  dispatch({ type: DELETE_APD_REQUEST });

  return axios
    .delete(`/apds/${id}`)
    .then(() => {
      dispatch({ type: DELETE_APD_SUCCESS });
      dispatch(fetch());
    })
    .catch(() => {
      dispatch({ type: DELETE_APD_FAILURE });
    });
};
