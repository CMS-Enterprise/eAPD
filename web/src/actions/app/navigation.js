import { push } from 'connected-react-router';
import { NAVIGATION_SCROLL_TO_WAYPOINT, RESET } from './symbols';

let jumping = false;

export const goToDashboard = ({ pushRoute = push } = {}) => dispatch => {
  dispatch({ type: RESET });
  dispatch(pushRoute('/'));
};

export const jumpTo = waypoint => dispatch => {
  if (waypoint) {
    jumping = true;
    setTimeout(() => {
      jumping = false;
    }, 100);
    dispatch({ type: NAVIGATION_SCROLL_TO_WAYPOINT, data: waypoint });
  }
};

export const scrollTo = waypoint => dispatch => {
  if (!jumping && waypoint) {
    dispatch({
      type: NAVIGATION_SCROLL_TO_WAYPOINT,
      data: waypoint
    });
  }
};
