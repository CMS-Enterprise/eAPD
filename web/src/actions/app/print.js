import { PRINT_APD } from './symbols';

export const printApd = ({ global = window } = {}) => dispatch => {
  dispatch({ type: PRINT_APD });
  global.print();
};

export default { printApd };
