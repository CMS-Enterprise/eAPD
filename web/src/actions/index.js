export const INCREMENT_COUNT = 'INCREMENT_COUNT';
export const DECREMENT_COUNT = 'DECREMENT_COUNT';

export const increment = () => ({ type: INCREMENT_COUNT });
export const decrement = () => ({ type: DECREMENT_COUNT });
