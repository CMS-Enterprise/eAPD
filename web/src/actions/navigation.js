export const NAVIGATION_SCROLL_TO_WAYPOINT = Symbol(
  'navigation : scroll to waypoint'
);

let jumping = false;

export const jumpTo = waypoint => dispatch => {
  if (waypoint) {
    jumping = true;
    setTimeout(() => {
      jumping = false;
    }, 100);
    dispatch({ type: NAVIGATION_SCROLL_TO_WAYPOINT, data: waypoint });
  }
};

export const scrollTo = waypoint => dispatch => {
  if (!jumping && waypoint) {
    dispatch({
      type: NAVIGATION_SCROLL_TO_WAYPOINT,
      data: waypoint
    });
  }
};
