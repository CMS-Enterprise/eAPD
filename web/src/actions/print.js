export const PRINT_APD = Symbol('print : apd');

export const printApd = ({ global = window } = {}) => dispatch => {
  dispatch({ type: PRINT_APD });
  global.print();
};
