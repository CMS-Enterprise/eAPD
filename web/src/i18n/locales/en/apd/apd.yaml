title: Program Summary
helpText: >-
  Provide an introduction for the state program. Include context for this funding request by providing information such as current size of program, future vision, and how the program is positioned within the state. This information is similar to the content from the IAPD executive summary template section.
overview:
  title: Program Overview
  helpText: >-
    Provide a brief, high level summary of the state program including history, and future vision. Include history and summary of future vision of the state program.
  reminder: >-

  yearsCovered: Choose the federal fiscal year(s) this APD covers.

introduction:
  title: Program Introduction
  helpText: >-
    Provide a brief, high level summary of the state program, history, and future vision.

hit:
  title: HIT Overview
  helpText: >-
    Summarize briefly planned HIT funded activities contained within this
    APD.
  reminder: >-

hie:
  title: HIE Overview
  helpText: >-
    Summarize planned HIE funded activities within this APD.

    (1) Description of the state's HIE approach (statewide, sub-state
    health information office (HIOs), etc.);

    (2) Anticipated risks and mitigation strategies;

    (3) The short and long term value proposition to providers;

    (4) The role of state government in governance and policy-setting.
  reminder: >-
    CMS recommends states connect to regional CMS HITECH contacts prior to developing HIE funding request. HIE funding requests are complex. Early consultation with CMS facilitates a more efficient approval process.
    (If there are no HIE activities, skip this section.)

mmis:
  title: MMIS Overview
  helpText: >-
    Briefly summarize planned MMIS funded activities contained within this APD.
  reminder: >-
    If there are no MMIS activities, skip this section.
