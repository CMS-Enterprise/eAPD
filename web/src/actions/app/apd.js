import { push } from 'connected-react-router';

import {
  CREATE_APD_FAILURE,
  CREATE_APD_REQUEST,
  CREATE_APD_SUCCESS,
  DELETE_APD_FAILURE,
  DELETE_APD_REQUEST,
  DELETE_APD_SUCCESS,
  FETCH_ALL_APDS_FAILURE,
  FETCH_ALL_APDS_REQUEST,
  FETCH_ALL_APDS_SUCCESS,
  SAVE_APD_FAILURE,
  SAVE_APD_REQUEST,
  SAVE_APD_SUCCESS,
  SELECT_APD,
  SET_APD_TO_SELECT_ON_LOAD
} from './symbols';
import { updateBudget } from '../budget';
import { EDIT_APD } from '../editApd/symbols';
import { ariaAnnounceApdLoaded, ariaAnnounceApdLoading } from '../aria';

import { selectApdData } from '../../reducers/apd.selectors';
import {
  selectHasChanges,
  selectPatches
} from '../../reducers/patch.selectors';
import { getIsAdmin } from '../../reducers/user.selector';

import axios from '../../util/api';
import initialAssurances from '../../util/regulations';

const LAST_APD_ID_STORAGE_KEY = 'last-apd-id';

export const saveApd = () => (dispatch, getState) => {
  const state = getState();
  const hasChanges = selectHasChanges(state);

  if (!hasChanges) {
    return Promise.resolve();
  }

  dispatch({ type: SAVE_APD_REQUEST });

  const { id: apdID } = selectApdData(state);
  const patches = selectPatches(state);

  return axios
    .patch(`/apds/${apdID}`, patches)
    .then(res => {
      dispatch({ type: SAVE_APD_SUCCESS, data: res.data });
      return res.data;
    })
    .catch(error => {
      if (error.response && error.response.status === 403) {
        dispatch({ type: SAVE_APD_FAILURE, data: 'save-apd.not-logged-in' });
      } else {
        dispatch({ type: SAVE_APD_FAILURE });
      }
      throw error;
    });
};

export const selectApd = (
  id,
  route,
  { global = window, pushRoute = push } = {}
) => dispatch => {
  dispatch(ariaAnnounceApdLoading());

  return axios.get(`/apds/${id}`).then(req => {
    dispatch({ type: SELECT_APD, apd: req.data });

    // By default, APDs get an empty object for federal citations. The canonical list of citations is in frontend
    // code, not backend. So if we get an APD with no federal citations, set its federal citations to the initial
    // values using an EDIT_APD action. That way the initial values get saved back to the API.
    if (Object.keys(req.data.federalCitations).length === 0) {
      dispatch({
        type: EDIT_APD,
        path: '/federalCitations',
        value: initialAssurances
      });
    }

    dispatch(updateBudget());
    dispatch(pushRoute(route));
    dispatch(ariaAnnounceApdLoaded());

    if (global.localStorage) {
      global.localStorage.setItem(LAST_APD_ID_STORAGE_KEY, id);
    }
  });
};

export const setApdToSelectOnLoad = () => (dispatch, getState) => {
  const isAdmin = getIsAdmin(getState());
  if (!isAdmin) {
    dispatch({ type: SET_APD_TO_SELECT_ON_LOAD });
  }
};

export const createApd = ({ pushRoute = push } = {}) => dispatch => {
  dispatch({ type: CREATE_APD_REQUEST });
  return axios
    .post('/apds')
    .then(async req => {
      dispatch({ type: CREATE_APD_SUCCESS, data: req.data });
      await dispatch(selectApd(req.data.id, '/apd', { pushRoute }));
    })
    .catch(error => {
      const reason = error.response ? error.response.data : 'N/A';
      dispatch({ type: CREATE_APD_FAILURE, data: reason });
    });
};

export const fetchAllApds = ({
  global = window,
  pushRoute = push,
  select = selectApd
} = {}) => (dispatch, getState) => {
  dispatch({ type: FETCH_ALL_APDS_REQUEST });

  const url = `/apds`;

  return axios
    .get(url)
    .then(async req => {
      const apd = Array.isArray(req.data) ? req.data : null;
      dispatch({
        type: FETCH_ALL_APDS_SUCCESS,
        data: apd.filter(({ status }) => status !== 'archived')
      });

      const {
        apd: { selectAPDOnLoad }
      } = getState();
      if (
        selectAPDOnLoad &&
        global.localStorage &&
        global.localStorage.getItem(LAST_APD_ID_STORAGE_KEY)
      ) {
        await dispatch(
          select(global.localStorage.getItem('last-apd-id'), {
            global,
            pushRoute
          })
        );
      }
    })
    .catch(error => {
      const reason = error.response ? error.response.data : 'N/A';
      dispatch({ type: FETCH_ALL_APDS_FAILURE, error: reason });
    });
};

export const deleteApd = (id, { fetch = fetchAllApds } = {}) => dispatch => {
  dispatch({ type: DELETE_APD_REQUEST });

  return axios
    .delete(`/apds/${id}`)
    .then(() => {
      dispatch({ type: DELETE_APD_SUCCESS });
      dispatch(fetch());
    })
    .catch(() => {
      dispatch({ type: DELETE_APD_FAILURE });
    });
};
