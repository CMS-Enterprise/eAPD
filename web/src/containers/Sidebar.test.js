import { shallow } from 'enzyme';
import React from 'react';
import sinon from 'sinon';

import {
  plain as Sidebar,
  mapStateToProps,
  mapDispatchToProps
} from './Sidebar';
import { saveApd } from '../actions/apd';
import { jumpTo } from '../actions/navigation';
import { printApd } from '../actions/print';

describe('Sidebar component', () => {
  const props = {
    activities: [
      { anchor: '#key1', key: 'key 1' },
      { anchor: '#key2', key: 'key 2' }
    ],
    activeSection: 'some section',
    jumpTo: sinon.spy(),
    place: { id: 'place id', name: 'place name' },
    printApd: sinon.spy(),
    saveApdToAPI: sinon.spy()
  };

  test('renders correctly', () => {
    expect(shallow(<Sidebar {...props} />)).toMatchSnapshot();
  });

  test('saves apd to api', () => {
    const saveApdToApiProp = sinon.spy();
    const component = shallow(
      <Sidebar {...props} saveApdToAPI={saveApdToApiProp} />
    );

    component.find('Btn[onClick][kind="primary"]').simulate('click');

    expect(saveApdToApiProp.called).toBeTruthy();
  });

  test('triggers a print event', () => {
    const printApdProp = sinon.spy();
    const component = shallow(<Sidebar {...props} printApd={printApdProp} />);

    component.find('Btn[onClick][kind="outline"]').simulate('click');

    expect(printApdProp.called).toBeTruthy();
  });

  test('maps state to props', () => {
    const state = {
      activities: {
        byKey: { key1: { name: 'activity 1' }, key2: { name: 'activity 2' } },
        allKeys: ['key1', 'key2']
      },
      navigation: {
        activeSection: 'where the runners are'
      }
    };

    expect(mapStateToProps(state)).toEqual({
      activities: [
        {
          key: 'key1',
          anchor: 'activity-key1',
          name: 'activity 1'
        },
        {
          key: 'key2',
          anchor: 'activity-key2',
          name: 'activity 2'
        }
      ],
      activeSection: 'where the runners are'
    });
  });

  test('maps dispatch to props', () => {
    expect(mapDispatchToProps).toEqual({
      jumpTo,
      printApd,
      saveApdToAPI: saveApd
    });
  });
});
