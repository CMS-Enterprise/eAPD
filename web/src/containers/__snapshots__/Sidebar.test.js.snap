// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component renders correctly 1`] = `
<div
  className="ds-l-col--3"
>
  <aside
    className="site-sidebar"
  >
    <div
      className="xs-hide sm-hide site-sidebar__sticky"
    >
      <div
        className="flex items-center ds-u-border-bottom--1 ds-u-padding-y--2 ds-u-margin-bottom--4"
      >
        <img
          alt="place name"
          className="align-middle mr2"
          height="40"
          src="/static/img/states/place id.svg"
          width="40"
        />
        <h1
          className="text-xl"
        >
          place name
        </h1>
      </div>
      <VerticalNav
        collapsed={false}
        items={
          Array [
            Object {
              "defaultCollapsed": true,
              "id": "apd-state-profile",
              "items": Array [
                Object {
                  "id": "apd-state-profile-overview",
                  "label": "Overview",
                  "onClick": [Function],
                  "url": "#apd-state-profile",
                },
                Object {
                  "id": "apd-state-profile-office",
                  "label": "State Director and Office Address",
                  "onClick": [Function],
                  "url": "#apd-state-profile-office",
                },
                Object {
                  "id": "apd-state-profile-key-personnel",
                  "label": "Key Personnel and Program Management",
                  "onClick": [Function],
                  "url": "#apd-state-profile-key-personnel",
                },
              ],
              "label": "Key State Personnel",
            },
            Object {
              "id": "apd-summary",
              "label": "Program Summary",
              "onClick": [Function],
              "url": "#apd-summary",
            },
            Object {
              "defaultCollapsed": true,
              "id": "prev-activities",
              "items": Array [
                Object {
                  "id": "prev-activities-overview",
                  "label": "Overview",
                  "onClick": [Function],
                  "url": "#prev-activities",
                },
                Object {
                  "id": "prev-activities-outline",
                  "label": "Prior Activities Overview",
                  "onClick": [Function],
                  "url": "#prev-activities-outline",
                },
                Object {
                  "id": "prev-activities-table",
                  "label": "Actual Costs",
                  "onClick": [Function],
                  "url": "#prev-activities-table",
                },
              ],
              "label": "Results of Previous Activities",
            },
            Object {
              "defaultCollapsed": true,
              "id": "activities",
              "items": Array [
                Object {
                  "id": "activities-overview",
                  "label": "Overview",
                  "onClick": [Function],
                  "url": "#activities",
                },
                Object {
                  "id": "activities-list",
                  "label": "Activity List",
                  "onClick": [Function],
                  "url": "#activities-list",
                },
                Object {
                  "id": "key 1",
                  "label": "Activity 1",
                  "onClick": [Function],
                  "url": "##key1",
                },
                Object {
                  "id": "key 2",
                  "label": "Activity 2",
                  "onClick": [Function],
                  "url": "##key2",
                },
              ],
              "label": "Program Activities",
            },
            Object {
              "id": "schedule-summary",
              "label": "Activity Schedule Summary",
              "onClick": [Function],
              "url": "#schedule-summary",
            },
            Object {
              "defaultCollapsed": true,
              "id": "budget",
              "items": Array [
                Object {
                  "id": "budget-overview",
                  "label": "Overview",
                  "onClick": [Function],
                  "url": "#budget",
                },
                Object {
                  "id": "budget-summary-table",
                  "label": "Summary Budget Table",
                  "onClick": [Function],
                  "url": "#budget-summary-table",
                },
                Object {
                  "id": "budget-federal-by-quarter",
                  "label": "Quarterly Federal Share",
                  "onClick": [Function],
                  "url": "#budget-federal-by-quarter",
                },
                Object {
                  "id": "budget-incentive-by-quarter",
                  "label": "Estimated Quarterly Incentive Payments",
                  "onClick": [Function],
                  "url": "#budget-incentive-by-quarter",
                },
              ],
              "label": "Proposed Budget",
            },
            Object {
              "id": "assurances-compliance",
              "label": "Assurances and Compliance",
              "onClick": [Function],
              "url": "#assurances-compliance",
            },
            Object {
              "defaultCollapsed": true,
              "id": "executive-summary",
              "items": Array [
                Object {
                  "id": "executive-summary-overview",
                  "label": "Overview",
                  "onClick": [Function],
                  "url": "#executive-summary",
                },
                Object {
                  "id": "executive-summary-summary",
                  "label": "Activities Summary",
                  "onClick": [Function],
                  "url": "#executive-summary-summary",
                },
                Object {
                  "id": "executive-summary-budget-table",
                  "label": "Program Budget Tables",
                  "onClick": [Function],
                  "url": "#executive-summary-budget-table",
                },
              ],
              "label": "Executive Summary",
            },
            Object {
              "id": "export-and-submit",
              "label": "Export and Submit",
              "onClick": [Function],
              "url": "#export-and-submit",
            },
          ]
        }
        selectedId="some section"
      />
      <div
        className="ds-u-margin-top--2"
      >
        <Btn
          extraCss={null}
          kind="primary"
          onClick={[Function]}
          size={null}
        >
          Save APD
        </Btn>
      </div>
    </div>
  </aside>
</div>
`;
